### PR TITLE
GAPI: Add BGR2RGB kernel to CPU backend.

### PR DESCRIPTION
**Pull request by anna-khakimova**
<cut/>

```
force_builders=Linux AVX2,Custom,Custom Win,Custom Mac,ARMv8,ARMv7,Linux32,Win32
disable_ipp:Custom=ON

buildworker:Custom=linux-3
build_image:Custom=ubuntu:18.04
CPU_BASELINE:Custom=AVX512_SKX

Xbuildworker:Custom=linux-1,linux-2,linux-4
Xbuild_image:Custom=powerpc64le
```

### CODE REVIEW

**Comment by anna-khakimova** [comment]
@dmatveev please take a look

File: modules/gapi/include/opencv2/gapi/imgproc.hpp [hunk 1]
```diff
@@ -871,6 +927,62 @@ Output image must be 8-bit unsigned 3-channel image @ref CV_8UC3.
...
+@sa BGR2I420
+*/
+GAPI_EXPORTS GMat I4202BGR(const GMat& src);
+
+/** @brief Converts an image from I420 color space to BGR color space.
+
+The function converts an input image from I420 color space to BGR.
+The conventional ranges for B, G, and R channel values are 0 to 255.
+
+Output image must be 8-bit unsigned 3-channel image input_height*2/3 @ref CV_8UC3.
```
Comment (dmatveev) [inline modules/gapi/include/opencv2/gapi/imgproc.hpp:977]: I'd rephrase text here and above on the image height. Currently it doesn't read good.

**Review by dmatveev — COMMENTED** [review]
Looks good but please fix comments

File: modules/gapi/include/opencv2/gapi/imgproc.hpp (hunk 1 above)
Comment (anna-khakimova) [reply modules/gapi/include/opencv2/gapi/imgproc.hpp]: Done.

**Comment by anna-khakimova** [comment]
@dmatveev please take a look. Comment was applied.

**Review by dmatveev — APPROVED** [review]

**Comment by anna-khakimova** [comment]
@alalek please merge

File: modules/gapi/include/opencv2/gapi/imgproc.hpp [hunk 2]
```diff
@@ -871,6 +927,70 @@ Output image must be 8-bit unsigned 3-channel image @ref CV_8UC3.
...
 GAPI_EXPORTS GMat RGB2YUV(const GMat& src);
 
+/** @brief Converts an image from BGR color space to I420 color space.
+
+The function converts an input image from BGR color space to I420.
+The conventional ranges for R, G, and B channel values are 0 to 255.
+
+Output image must be 8-bit unsigned 1-channel image. @ref CV_8UC1.
+Width of I420 output image must be the same as width of input image.
+Height of I420 output image must be equal 3/2 from height of input image.
```
Comment (alalek) [inline modules/gapi/include/opencv2/gapi/imgproc.hpp:937]: > 3/2 from height

Why is not tuple of `Y` and `UV`?

/cc @TolyaTalamanov (implemented drawing over NV12 frames)

Comment (dmatveev) [reply modules/gapi/include/opencv2/gapi/imgproc.hpp:937]: @alalek tuple of Y and UV is what we'll do next, but the current request is to provide a `cvtColor`-based counterpart.

File: modules/gapi/include/opencv2/gapi/imgproc.hpp [hunk 3]
```diff
@@ -812,6 +854,20 @@ The algorithm normalizes the brightness and increases the contrast of the image.
  */
 GAPI_EXPORTS GMat equalizeHist(const GMat& src);
 
+/** @brief Converts an image from BGR color space to RGB color space.
+
+The function converts an input image from BGR color space to RGB.
+The conventional ranges for B, G, and R channel values are 0 to 255.
+
+Output image must be 8-bit unsigned 3-channel image @ref CV_8UC3.
```
[the remark this answers is not in the document]
Comment (dmatveev) [reply modules/gapi/include/opencv2/gapi/imgproc.hpp:862]: +1, probably simple `is` is enough

File: modules/gapi/include/opencv2/gapi/imgproc.hpp (hunk 2 above)
Comment (alalek) [reply modules/gapi/include/opencv2/gapi/imgproc.hpp:937]: Please note that there is no C++ overloads for return type only. So it will be hard to improve that without breaking of existed code.
Just keep in mind.

Comment (anna-khakimova) [reply modules/gapi/include/opencv2/gapi/imgproc.hpp:937]: ok

File: modules/gapi/include/opencv2/gapi/imgproc.hpp (hunk 3 above)
[the remark this answers is not in the document]
Comment (anna-khakimova) [reply modules/gapi/include/opencv2/gapi/imgproc.hpp:862]: @alalek Half of the comments in this file phrased as "Output image must be 8-bit unsigned 3-channel image ". For example for  YUV2RGB kernel:
** @brief Converts an image from YUV color space to RGB.
The function converts an input image from YUV color space to RGB.
The conventional ranges for Y, U, and V channel values are 0 to 255.

Output image must be 8-bit unsigned 3-channel image @ref CV_8UC3.

@note Function textual ID is "org.opencv.imgproc.colorconvert.yuv2rgb"

@param src input image: 8-bit unsigned 3-channel image @ref CV_8UC3.

@sa RGB2Lab, RGB2YUV
*/
GAPI_EXPORTS GMat YUV2RGB(const GMat& src);

However, for some reason, attention was paid only to my comment. The remaining 14 color conversion kernels will remain with comments formulated as "Output image must be 8-bit unsigned ...". I thought we were fighting for uniformity in the comments. But apparently I was wrong.

[the remark this answers is not in the document]
Comment (anna-khakimova) [reply modules/gapi/include/opencv2/gapi/imgproc.hpp]: Done
